# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/named_pipe_policy_test.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/named_pipe_policy_test.cc
--- a/security/sandbox/chromium/sandbox/win/src/named_pipe_policy_test.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/named_pipe_policy_test.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -5,26 +5,20 @@
 #include "base/win/windows_version.h"
 #include "sandbox/win/src/handle_closer.h"
 #include "sandbox/win/src/sandbox.h"
+#include "sandbox/win/src/sandbox_factory.h"
 #include "sandbox/win/src/sandbox_policy.h"
-#include "sandbox/win/src/sandbox_factory.h"
 #include "sandbox/win/tests/common/controller.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
 namespace sandbox {
 
-
-SBOX_TESTS_COMMAND int NamedPipe_Create(int argc, wchar_t **argv) {
-  if (argc < 1 || argc > 2) {
-    return SBOX_TEST_FAILED_TO_EXECUTE_COMMAND;
-  }
-  if ((NULL == argv) || (NULL == argv[0])) {
+SBOX_TESTS_COMMAND int NamedPipe_Create(int argc, wchar_t** argv) {
+  if (argc < 1 || argc > 2 || !argv || !argv[0])
     return SBOX_TEST_FAILED_TO_EXECUTE_COMMAND;
-  }
 
-  HANDLE pipe = ::CreateNamedPipeW(argv[0],
-                                   PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,
-                                   PIPE_TYPE_BYTE | PIPE_READMODE_BYTE, 1, 4096,
-                                   4096, 2000, NULL);
+  HANDLE pipe = ::CreateNamedPipeW(
+      argv[0], PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,
+      PIPE_TYPE_BYTE | PIPE_READMODE_BYTE, 1, 4096, 4096, 2000, nullptr);
   if (INVALID_HANDLE_VALUE == pipe)
     return SBOX_TEST_DENIED;
 
@@ -41,14 +35,13 @@
   }
 
   OVERLAPPED overlapped = {0};
-  overlapped.hEvent = ::CreateEvent(NULL, TRUE, TRUE, NULL);
-  BOOL result = ::ConnectNamedPipe(pipe, &overlapped);
+  overlapped.hEvent = ::CreateEvent(nullptr, true, true, nullptr);
+  bool result = ::ConnectNamedPipe(pipe, &overlapped);
 
   if (!result) {
     DWORD error = ::GetLastError();
-    if (ERROR_PIPE_CONNECTED != error &&
-        ERROR_IO_PENDING != error) {
-          return SBOX_TEST_FAILED;
+    if (ERROR_PIPE_CONNECTED != error && ERROR_IO_PENDING != error) {
+      return SBOX_TEST_FAILED;
     }
   }
 
@@ -82,7 +75,7 @@
   // namedpipe name. Here we apply it like a wildcard. http://b/893603
   EXPECT_TRUE(runner.AddRule(TargetPolicy::SUBSYS_NAMED_PIPES,
                              TargetPolicy::NAMEDPIPES_ALLOW_ANY,
-                              L"\\\\.\\pipe\\test*"));
+                             L"\\\\.\\pipe\\test*"));
 
   EXPECT_EQ(SBOX_TEST_DENIED,
             runner.RunTest(L"NamedPipe_Create \\\\.\\pipe\\test\\..\\bleh"));
@@ -101,8 +94,8 @@
   // disable all string parsing and to send the string that follows it straight
   // to the file system."
   // http://msdn.microsoft.com/en-us/library/aa365247(VS.85).aspx
-  const wchar_t* argv[2] = { L"\\\\?\\pipe\\test\\..\\bleh",
-                             L"\\Device\\NamedPipe\\test" };
+  const wchar_t* argv[2] = {L"\\\\?\\pipe\\test\\..\\bleh",
+                            L"\\Device\\NamedPipe\\test"};
   EXPECT_EQ(SBOX_TEST_SUCCEEDED,
             NamedPipe_Create(2, const_cast<wchar_t**>(argv)));
 }
@@ -116,7 +109,7 @@
   // namedpipe name. Here we apply it like a wildcard. http://b/893603
   EXPECT_TRUE(runner.AddRule(TargetPolicy::SUBSYS_NAMED_PIPES,
                              TargetPolicy::NAMEDPIPES_ALLOW_ANY,
-                              L"\\\\.\\pipe\\test*"));
+                             L"\\\\.\\pipe\\test*"));
 
   EXPECT_EQ(SBOX_TEST_SUCCEEDED,
             runner.RunTest(L"NamedPipe_Create \\\\.\\pipe\\testbleh"));